phpBB/phpbb/content_visibility.php

```diff
@@ -216,7 +222,13 @@ public function get_visibility_sql($mode, $forum_id, $table_alias = '')
 		}
 		else
 		{
-			$where_sql .= $table_alias . $mode . '_visibility = ' . ITEM_APPROVED;
+			$field_name = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
+			$visibility_query = $table_alias . $mode . '_visibility = ';
+
+			$where_sql .= '(' . $visibility_query . ITEM_APPROVED . ')';
+			$where_sql .= ' OR (';
+			$where_sql .= '(' . $visibility_query . ITEM_UNAPPROVED . ' OR ' . $visibility_query . ITEM_REAPPROVE . ')';
+			$where_sql .= ' AND ' . $table_alias . $field_name . ' = ' . ((int) $this->user->data['user_id']) . ')';
```

Comment:
May you please change this to the [query builder](https://github.com/phpbb/phpbb/pull/4006) to allow extensions to more easily change this query for their own needs?

Reply:
This query is already using query builder (in the calling function), here we are just building a WHERE clause for query builder.

Reply:
@v12mike query builder: https://github.com/phpbb/phpbb/pull/4006
https://area51.phpbb.com/phpBB/viewtopic.php?t=47306

Reply:
There is an existing event 'core.phpbb_content_visibility_get_visibility_sql_before' in the middle of building the sql, which I think would get broken  if the sql builder was used more extensively here.  Can you see how to do it?

Reply:
Good point. I failed to see it.
It's sad but...
I rest my case.